frontend/src/admin/components/dropdown-menu/dropdown-menu.tsx

```diff
@@ -0,0 +1,36 @@
+import React, {useState} from "react";
+import DropdownMenuIcon from "../dropdown-menu-icon/dropdown-menu-icon";
+import "./dropdown-menu.scss";
+
+interface DropdownMenuProps {
+    children: React.ReactNode
+}
+
+export default function DropdownMenu( {children}: DropdownMenuProps) {
+    const [showDropdown, setShowDropdown] = useState(false);
+
+    function toggleDropdown() {
+        setShowDropdown(!showDropdown);
+    }
+
+    function hideDropdown(event: React.FocusEvent<HTMLButtonElement>): void {
+        if (event.currentTarget !== event.target) {
+            return
+        }
+        setShowDropdown(false);
+    }
+
+    return (
+        <div className={`c-dropdown-menu${showDropdown ? '--active' : ''}`}>
+            <button
+                onClick={toggleDropdown}
+                onBlur={(e: React.FocusEvent<HTMLButtonElement>): void => hideDropdown(e)}
```

Comment:
The types are not necessary and try to avoid abbreviations. If you want to reduce it as much as possible, then you can simple give the `hideDropdown` as argument to onBlur, since onBlur and hideDropdown have matching type definitions.